### PR TITLE
feat(ios): be able to detect an emulated iOS app on Apple Silicon

### PR DESCRIPTION
**Pull request by hansemannn**
Example code to test (compile an app and run it on an Apple Silicon mac, e.g. via Testflight):
```js
const win = Ti.UI.createWindow({
    backgroundColor: '#fff'
});

const btn = Ti.UI.createButton({
    title: 'Check if emulated'
});

btn.addEventListener('click', () => {
    alert(`Is emulated iOS app on Apple Silicon: ${Ti.Platform.isAppleSiliconTranslated}`);
});

win.add(btn);
win.open();
```

### CODE REVIEW

**Comment by hansemannn** [comment]
@janvennemann Would you mind checking the proxy part? The logic itself is from Apple :)

File: iphone/Classes/PlatformModule.m [hunk 1]
```diff
@@ -256,6 +256,21 @@ - (NSString *)identifierForAdvertising
 }
 #endif
 
+- (bool)isRunningOnAppleSilicon
```
Comment (janvennemann) [inline iphone/Classes/PlatformModule.m:259]: Maybe name it `isTranslatedBinaryOnAppleSilicon` to indicate that it refers to the Rosetta translation and not just whether the App is running on an Apple Silicon device?

File: apidoc/Titanium/Platform/Platform.yml [hunk 2]
```diff
@@ -271,6 +271,18 @@ properties:
     permission: read-only
     since: "7.0.0"
 
+  - name: isRunningOnAppleSilicon
+    summary: |
+        A Boolean value that indicates whether the current apps is emulated
+        via an Apple Silicon device.
```
Comment (janvennemann) [inline apidoc/Titanium/Platform/Platform.yml:277]:
```suggestion
        A Boolean value that indicates whether the current app is running as a translated
        binary using Rosetta on an Apple Silicon device.
```

**Review by janvennemann — CHANGES_REQUESTED** [review]
Just some small notes regarding naming and the method summary in the docs.

I'm also wondering if it would make sense to have this in an iOS only namespace? The is `Ti.Platform.Android` for Android specific device infos and this likely falls in the same category, just for iOS, right?

**Review by janvennemann — APPROVED** [review]
Thanks! Yeah, makes sense to only go through the effort of creating the new namespace when there are actually more iOS only utils.

**Pull request merged**